LibreNMS/Alert/RunAlerts.php

```diff
@@ -131,6 +132,23 @@ public function describeAlert($alert)
         }
         $extra = $alert['details'];
 
+        $obj['applications'] = $device->applications->groupBy('app_type');
+        $obj['applications_metrics'] = [];
```

Comment:
$obj['applications'] = $device->applications->load('metrics')->groupBy('app_type');

Reply:
@murrant This is actually a fairly bad solution as it creates a mess that then needs to be processed.

It creates a second un-needed array that helps no one. On top of now having to use `[0]` you now also need unneeded array of `[metrics]`... which now you need to read through to find the one you want.

So lets say you want to display `iax2_iax2unmonitored` for `asterisk`, you need to read through `[asterisk][0][metrics]` till you find the array that has the metric you want.

Reply:
negative.  you should never use [0] that will only show the first service. it should be
```blade
@foreach($alert->applications as $app_name => $app)
    <h1>{{ $app_name }}</h1>
    State: {{ $app->app_state }}
    Metrics:
    @foreach($app->metrics as $metric) 
        {{ $metric->metric }}: {{ $metric->value }}
    @endforeach
@endforeach

Reply:
@murrant No.

There is no good reason to use anything other than `[0]` unless we are talking about an outlier. Look at the code for apps. Only two will ever have more than one and it is legacy. Anything not drdb or proxmox has it's instances stuff handled via app data.

On top of that you are also not going to want to really be dumping metrics like that for a alert. It is pure information spam.

We should try to make writing alert templates as straight forward as possible, which means transforming the metrics to a associative array.

Reply:
Let me put it this way. Dumping info like that in alerts is just going to mean a bad day for every one.

For example I have some apps that have well over a thousand different metrics.

Look at the new Suricata stuff for example. That generates over a bit over 900 metrics for a single instance(instance here being managed via app data).

Reply:
Making an assumption that only one app exists is going to cause bugs where users cannot access their 2nd app.

My template was an example of how it would be access, not a suggested template. Dumping 900 metrics in a notification like that would indeed be stupid.  But that doesn't help the issue that an alert can have multiple applications.